server/game/core/cardSelector/BaseCardSelector.js

```diff
@@ -135,7 +136,10 @@ class BaseCardSelector {
         if (!EnumHelpers.cardZoneMatches(card.zoneName, this.zoneFilter) && card.zoneName !== ZoneName.Capture) {
             return false;
         }
-        if (card.zoneName === ZoneName.Hand && card.controller !== choosingPlayer) {
+        if ((this.discardingPlayerType == null || this.discardingPlayerType === RelativePlayer.Self) && card.zoneName === ZoneName.Hand && card.controller !== choosingPlayer) {
```

Comment:
I'm thinking on this some more again - do we actually even need this checking logic at all? I'm not sure why hand would need a specific guard that other zones like resources wouldn't. It seems like this was originally set up to block cards from the opponent's hand even if the card filters were set up to allow it. Would things work if we just remove this altogether and rely on the zone filters to get set up right?

Reply:
Agreed. I was able to get rid of this by adding back missing controller: RelativePlayer.Self to various cards to not require this check